src/android/Geolocation.java

```diff
@@ -128,50 +193,58 @@ public void onRequestPermissionResult(int requestCode, String[] permissions, int
+    private void getLocationGms(LocationContext locationContext) {
+        JSONArray args = locationContext.getExecuteArgs();
+        long timeout = args.optLong(2);
+        boolean enableHighAccuracy = args.optBoolean(0, false);
+        LocationRequest request = LocationRequest.create();
 
-        if(status != ConnectionResult.SUCCESS) {
-            PluginResult result;
+        request.setNumUpdates(1);
 
-            if(googleApiAvailability.isUserResolvableError(status)) {
-                googleApiAvailability.getErrorDialog(cordova.getActivity(), status, 1).show();
-                result = new PluginResult(PluginResult.Status.ERROR, LocationError.GOOGLE_SERVICES_ERROR_RESOLVABLE.toJSON());
-            }
-            else {
-                result = new PluginResult(PluginResult.Status.ERROR, LocationError.GOOGLE_SERVICES_ERROR.toJSON());
-            }
+        // This is necessary to be able to get a response when location services are initially off and then turned on before this request.
+        request.setInterval(0);
+
+        if(enableHighAccuracy) {
+            request.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+        }
 
-            callbackContext.sendPluginResult(result);
-            return false;
+        if(timeout != 0) {
+            request.setExpirationDuration(timeout);
         }
 
-        return true;
+        requestLocationUpdatesIfSettingsSatisfiedGms(locationContext, request);
     }
 
-    private void getLocation(LocationContext locationContext) {
+    /**
+     * Get Location from Huawei Devices / No Google Services inside
+     * @param locationContext the locationContext
+     */
+    private void getLocationHms(LocationContext locationContext) {
         JSONArray args = locationContext.getExecuteArgs();
         long timeout = args.optLong(2);
         boolean enableHighAccuracy = args.optBoolean(0, false);
-        LocationRequest request = LocationRequest.create();
+        com.huawei.hms.location.LocationRequest request = com.huawei.hms.location.LocationRequest.create();
 
         request.setNumUpdates(1);
 
         // This is necessary to be able to get a response when location services are initially off and then turned on before this request.
         request.setInterval(0);
 
         if(enableHighAccuracy) {
-            request.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+            request.setPriority(com.huawei.hms.location.LocationRequest.PRIORITY_HIGH_ACCURACY);
         }
 
         if(timeout != 0) {
             request.setExpirationDuration(timeout);
         }
 
-        requestLocationUpdatesIfSettingsSatisfied(locationContext, request);
+        requestLocationUpdatesIfSettingsSatisfiedHms(locationContext, request);
     }
 
+    /**
+     * Add Watch Locations in devices which contains google services
+     * @param locationContext the locationContext
+     */
     private void addWatch(LocationContext locationContext) {
```

Comment:
In order to follow the same approach as other methods, shouldn't this be changed to `addWatchGms`? 

However, most of the code inside the `addWatch/addWatchHms` methods is identical besides one or two property/methods. Can't we reuse code as much as possible in order to avoid too much replication?